File: Algorithm_7_Reverse_Integer.cpp
#include <iostream>
#include <climits>
using namespace std;
long long int reverse(long long int x);
int main()
{
	long long int x=1534236469;
	cout<<reverse(x);
	return 0;
}
long long int reverse(long long int x) {
    long long int res=0;
    while (x!=0)
    {
        res=res*10+x%10;
        x/=10;
	}
	if (res<INT_MIN || res>INT_MAX) res=0;
	if (x<0) res=-res;
    return res;
}

File: Algorithm_7_Reverse Integer.java
// 思路：循环把x对10取模的结果加到res上。通过判断(res - x % 10) / 10 != lastRes是否成立来知道是否发生溢出
// 不需要考虑x的政府，因为while中的条件用的是x != 0而不是x > 0

class Solution {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int lastRes = res;
            res = res * 10 + x % 10;
            if ((res - x % 10) / 10 != lastRes) { // an overflow happens
                return 0;
            }
            x /= 10;
        }
        return res;
    }
}